### PR TITLE
Fixes for yupgi_alert

### PR DESCRIPTION
**Pull request by jrb0001**
These changes are needed to compile https://github.com/forcecore/yupgi_alert0.

See also: https://github.com/forcecore/yupgi_alert0/pull/87.

### CODE REVIEW

**Comment by forcecore** [comment]
Looks my changes for aircraft carrier logic is making its way inside upstream code :)

https://www.youtube.com/watch?v=H1iBoTokt-k : Demo

There are more changes though in landing code that allows those spawned ones to land "midair".

File: OpenRA.Mods.Common/Activities/Enter.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #region Copyright & License Information
 /*
- * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
```
Comment (pchote) [inline OpenRA.Mods.Common/Activities/Enter.cs]: Most of the changes in this file appear to be reverting #12601.  The downstream code needs to be updated instead.

Comment (jrb0001) [inline OpenRA.Mods.Common/Activities/Enter.cs]: Checked out the wrong branch while copying over changes, should be fixed now

**Comment by jrb0001** [comment]
@forcecore It is just to make it compile. You should file a separate PR with the aircraft carrier changes.

**Comment by atlimit8** [comment]
Please squash the commits into one.

**Comment by jrb0001** [comment]
@atlimit8 done

**Comment by pchote** [comment]
This PR poses a conundrum given the discussion in https://github.com/forcecore/yupgi_alert0/pull/87 and now @GraionDilach's engine fork.

On the one hand I think its unreasonable for us to force community mods to fork everything that they want to modify, but on the other hand I don't like the idea of us making upstream changes for the sake of compatibility with code that has design issues that would prevent it from being merged upstream as-is.  This is a topic that we have not yet given serious thought to.

For this PR specifically, looking at https://github.com/forcecore/yupgi_alert0/pull/87 I would expect `AIDeployHelper` to call `ResolveOrder` instead of reaching directly into the internals of `DeployToUpgrade`, and `EnterSpawner` should be working with info passed into the virtual methods (which may require some changes to the method signatures).

The problem with making the internals of these classes less private is that, it might (a) be reverted in the future as a code style cleanup by somebody who doesn't know it's being used downstream, or (b) prevent us from making necessary changes because we're worried about breaking downstream code.

**Comment by forcecore** [comment]
For (b) (and possibly (a)) I'd say, let it break. I think the main reason that OpenRA moved from Mod selector to Game.Mod style of launch is to let breaking happen and let the mods update their mods on their own schedule.

**Comment by GraionDilach** [comment]
>For this PR specifically, looking at forcecore/yupgi_alert0#87 I would expect AIDeployHelper to call ResolveOrder instead of reaching directly into the internals of DeployToUpgrade

This is something I've written a PR already for but seems it got lost during a refactor already (although true that my PR back then broke undeployable actors, something I overlooked back then and only fixed earlier today).

Actually, this ticket was among the main reasons why I forked, because this ticket + related ultimately would have forced Mods.AS into a dependency of yupgi_alert, preventing me on AS side to take code from it due to namespace conflicts thenafter. (To stop the misassumptions I've noticed in the IRC logs, the other ones were the already-mentioned actor variations possibility in https://github.com/OpenRA/OpenRA/issues/12554#issuecomment-271105904, and my bad experiences in regards to #11624 and #11311. My personal development habits has nothing to do with my decision to softfork.)

I actually consider this PR aggressive in terms even, because it tried to adjust three projects for one person's workflow.

**Comment by pchote** [comment]
https://github.com/forcecore/yupgi_alert0/pull/87 has been closed, so it no longer makes sense to take this PR.

**Pull request closed without merging**